File: csharp/01-lasagna.cs
class Lasagna
{
    private const int PreparationPerLayer = 2;

    public int ExpectedMinutesInOven(){
        return 40;
    }

    public int RemainingMinutesInOven(int minutesInOven){
        return ExpectedMinutesInOven() - minutesInOven;
    }

    public int PreparationTimeInMinutes(int layers){
        return PreparationPerLayer * layers;
    }

    public int ElapsedTimeInMinutes(int layers, int minutesInOven){
        return PreparationTimeInMinutes(layers) + minutesInOven;
    }
}

File: java/01-lasagna.java
public class Lasagna {

    private static final int PREPARATION_PER_LAYER = 2;  

    
    public int expectedMinutesInOven(){
        return 40;
    }


    public int remainingMinutesInOven(int minutesInOven){
        return expectedMinutesInOven() - minutesInOven;
    }


    public int preparationTimeInMinutes(int layers){
        return layers * PREPARATION_PER_LAYER;
    }

    public int totalTimeInMinutes(int layers, int minutesInOven){
        return preparationTimeInMinutes(layers) + minutesInOven;
    }
}